tensorboard/webapp/app_routing/effects/app_routing_effects.ts

```diff
@@ -69,13 +73,29 @@ export class AppRoutingEffects {
     ofType(navigationRequested)
   );
 
+  private getPathPrefixlessPathname(pathname: string): string {
+    if (this.pathPrefix !== null && pathname.startsWith(this.pathPrefix)) {
+      return pathname.slice(this.pathPrefix.length);
+    }
+    return pathname;
+  }
+
   private readonly onInit$: Observable<Navigation> = this.actions$
     .pipe(ofType(initAction))
     .pipe(
       delay(0),
+      switchMap(() => {
+        return this.store.select(getPathPrefixConfig).pipe(
+          filter((pathPrefix) => pathPrefix !== null),
+          take(1),
+          tap((pathPrefix) => {
+            this.pathPrefix = pathPrefix;
```

Comment:
Would things be more maintainable if we didn't store pathPrefix on AppRoutingEffects, but made the Location's getPath, onPopState, replaceState methods aware of it?  e.g.
```ts
this.location.setPathPrefix(pathPrefix);
```

I think Location's method surface is much smaller to check than making sure every usage of `this.location.getPath()` is covered in this file.  In case some other file in app_routing needed to inject Location, it wouldn't need to do all this wrapping as well.

Reply:
I thought about that but was a bit weary of putting business logic into what seems like a clear API, location!

According to our usages of the state, we need to be able to
- read synchronously
- discern was modified 

Those one or two are separate APIs that really do not belong in the Location, imo.